src/SharedFileList.cpp

```diff
@@ -745,7 +745,7 @@ void CSharedFileList::SendListToServer(){
 	//   - this function is called once when connecting to a server and when a file becomes shareable - so, it's called rarely.
 	//   - if the compressed size is still >= the original size, we send the uncompressed packet
 	// therefor we always try to compress the packet
-	if (server->GetTCPFlags() & SRV_TCPFLG_COMPRESSION){
+	if (server && (server->GetTCPFlags() & SRV_TCPFLG_COMPRESSION)){
```

Comment:
First of all, this function should only be called when we're connected to a server. But just in case, if we are not connected (i.e. `GetCurrentServer()` returns `NULL` on line 707) we could bail out immediately, because `SendPacket()` will fail anyway at the end of the function. Of course that would render all further checks for `server` being non-null obsolete.